packages/navigator/src/browser/navigator-widget.ts

```diff
@@ -79,21 +79,56 @@ export class FileNavigatorWidget extends FileTreeWidget {
                         this.model.expandNode(child);
                     }
                 }
+            }),
+            this.workspaceService.onWorkspaceRootChanged(async event => {
+                if (event) {
+                    await this.refreshWidget(event.root, event.roots, event.workspaceConfigFile);
+                }
             })
         ]);
     }
 
-    protected initialize(): void {
-        this.workspaceService.root.then(async resolvedRoot => {
-            if (resolvedRoot) {
-                const uri = new URI(resolvedRoot.uri);
+    protected async initialize(): Promise<void> {
+        const workspaceChangeData = await this.workspaceService.workspaceRootChange;
+        await this.refreshWidget(workspaceChangeData.root, workspaceChangeData.roots, workspaceChangeData.workspaceConfigFile);
+    }
+
+    protected async refreshWidget(root: FileStat | undefined, roots: FileStat[], workspaceConfig: FileStat | undefined) {
+        if (root) {
+            if (roots.length === 1) { // only one folder is opened
+                this.model.hasMultipleRoots = false;
+                const uri = new URI(root.uri);
                 const label = this.labelProvider.getName(uri);
-                const icon = await this.labelProvider.getIcon(resolvedRoot);
-                this.model.root = DirNode.createRoot(resolvedRoot, label, icon);
-            } else {
-                this.update();
+                const icon = await this.labelProvider.getIcon(root);
+                this.model.root = DirNode.createRoot(root, label, icon);
+            } else if (roots.length > 1) { // more than one folder
```

Comment:
Ideally having a single tree workspace is just one scenario that multi-root workspace supports. 
I actually don't have to have 
```ts
if (roots.length === 1) { // only one folder is opened
...
else if (roots.length > 1) { // more than one folder
```
And the logic under the `else if` should be good enough to have the file tree displayed in both cases. 
 
The reason that I am separating the `one-root` and `multi-root` is that, the file tree would collapse due to https://github.com/theia-ide/theia/issues/1781 and the tree wouldn't look as nice as it should, in the multi-root setting.
I will keep the `single root` and `multi root` display logic separated until #1781 is fixed.